arch/x86/chunkset_avx.c

```diff
@@ -45,6 +83,51 @@ static inline void storechunk(uint8_t *out, chunk_t *chunk) {
     _mm256_storeu_si256((__m256i *)out, *chunk);
 }
 
+static inline chunk_t GET_CHUNK_MAG(uint8_t *buf, uint32_t *chunk_rem, uint32_t dist) {
+    lut_rem_pair lut_rem = perm_idx_lut[dist - 3];
+    __m256i ret_vec;
+    /* While technically we only need to read 4 or 8 bytes into this vector register
+     * for a lot of cases, GCC is compiling this to a shared load for all branches,
+     * preferring the simpler code.  Given that the buf value isn't in GPRs to begin with,
+     * the 256 bit load is _probably_ just as inexpensive */
+    *chunk_rem = lut_rem.remval;
+
+#ifdef Z_MEMORY_SANITIZER
+    /* See note in chunkset_sse4.c for why this is ok */
+    __msan_unpoison(buf+dist, 32 - dist);
+#endif
+
+    if (dist < 16) {
+        /* This simpler case still requires us to shuffle in a 128 bit lanes, so we must
+         * apply a static offset after broadcasting the first vector register to both
```

Comment:
Short line with sentence spanning to next line... We use 120 characters per line.